analysis/src/Protocol.ml

```diff
@@ -126,3 +132,17 @@ let stringifyCodeAction ca =
   Printf.sprintf {|{"title": "%s", "kind": "%s", "edit": %s}|} ca.title
     (codeActionKindToString ca.codeActionKind)
     (ca.edit |> stringifyCodeActionEdit)
+
+let stringifyDiagnostic d =
+  Printf.sprintf {|{
+  "range": %s,
+  "message": "%s",
+  "severity": %d,
+  "source": "ReScript"
```

Comment:
Is this needed by the server? Seems redundant to send this info all the time given we control commands on both ends.
Unless it's a little convenience to avoid doing work on the server.